huaweicloud/services/mpc/resource_huaweicloud_mpc_transcoding_template.go

```diff
@@ -0,0 +1,503 @@
+		Level:              utils.Int32(int32(video["level"].(int))),
+		Preset:             utils.Int32(int32(video["preset"].(int))),
+		RefFramesCount:     utils.Int32(int32(video["ref_frames_count"].(int))),
+		MaxIframesInterval: utils.Int32(int32(video["max_iframes_interval"].(int))),
+		BframesCount:       utils.Int32(int32(video["bframes_count"].(int))),
+		FrameRate:          utils.Int32(int32(video["frame_rate"].(int))),
+		Width:              utils.Int32(int32(video["width"].(int))),
+		Height:             utils.Int32(int32(video["height"].(int))),
+		BlackCut:           utils.Int32(int32(video["black_cut"].(int))),
+		OutputPolicy:       buildVideoOutputPolicyOpts(video["output_policy"].(string)),
+	}
+
+	return &videoOpts
+}
+
+func buildVideoOutputPolicyOpts(outputPolicy string) *mpc.VideoOutputPolicy {
+	if outputPolicy == "" {
+		return nil
+	}
+
+	var outputPolicyOpts mpc.VideoOutputPolicy
+	switch outputPolicy {
+	case "discard":
+		outputPolicyOpts = mpc.GetVideoOutputPolicyEnum().DISCARD
+	case "transcode":
+		outputPolicyOpts = mpc.GetVideoOutputPolicyEnum().TRANSCODE
+	default:
+		log.Printf("[WARN] output_policy invalid: %s", outputPolicy)
+		return nil
+	}
+
+	return &outputPolicyOpts
+}
+
+func resourceMPCTranscodingTemplateCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.HcMpcV1Client(config.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating MPC client : %s", err)
+	}
+
+	createOpts := mpc.TransTemplate{
+		TemplateName: d.Get("name").(string),
+		Video:        buildVideoOpts(d.Get("video").([]interface{})),
+		Audio:        buildAudioOpts(d.Get("audio").([]interface{})),
+		Common:       buildCommonOpts(d.Get("common").([]interface{})),
+	}
+
+	createReq := mpc.CreateTransTemplateRequest{
+		Body: &createOpts,
+	}
+
+	resp, err := client.CreateTransTemplate(&createReq)
+	if err != nil {
+		return diag.Errorf("error creating MPC transcoding template: %s", err)
+	}
+
+	d.SetId(strconv.FormatInt(int64(*resp.TemplateId), 10))
+
+	return resourceMPCTranscodingTemplateRead(ctx, d, meta)
+}
+
+func resourceMPCTranscodingTemplateRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.HcMpcV1Client(config.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating MPC client : %s", err)
+	}
+
+	id, err := strconv.ParseInt(d.Id(), 10, 32)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	resp, err := client.ListTemplate(&mpc.ListTemplateRequest{TemplateId: &[]int32{int32(id)}})
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving MPC transcoding template")
+	}
+
+	templateList := *resp.TemplateArray
+	template := templateList[0].Template
+	if template == nil {
+		log.Printf("unable to retrieve MPC transcoding template: %s", d.Id())
+		d.SetId("")
+		return nil
+	}
+
```

Comment:
If the length of the returned result is 0, panic will be triggered: `invalid array index 0`.

Reply:
I have tested it, the array is never empty.

Reply:
It's a risk, we cannot guarantee that future modifications on the service side will cause this panic to occur.
suggest to add a pre-check.